src/lib/MessageContainer.svelte

```diff
@@ -14,28 +23,42 @@
     export const getLast = () => messages[messages.length - 1];
 
     export const addMessage = (message: MessageType) => {
-        messages.push(message);
-        messages = messages;
+        addMessageToStore(message);
+        // messages.push(message);
+        // messages = messages;
         scroll();
         return message;
     };
 
     export const updateLastMessageContent = (content: string) => {
-        messages[messages.length - 1].content = content;
-        lastMessage?.update(content);
+        updateLastMsgInStore(content);
+        // messages[messages.length - 1].content = content;
+        // lastMessage?.update(content);
         scroll();
     };
 
     export const appendLastMessageContent = (content: string) => {
-        getLast().content += content;
-        lastMessage?.append(content);
+        appendContentToLastMsgInStore(content);
+        // getLast().content += content;
+        // lastMessage?.append(content);
         scroll();
     };
 
     export const clearMessages = () => {
-        messages = [];
+        resetStore();
+        // messages = [];
         scroll();
     };
+
+    onMount(() => {
+        // Subscribe to changes in the messageStore
+        const unsubscribe = messageStore.subscribe((data) => {
+            messages = data;
```

Comment:
Something cool about svelte stores is you can actually use them directly and they will be fully "reactive". This means you can get rid of the messages variable directly, and instead reference the messages array as `$messageStore`.

Check out this PR: https://github.com/mitmedialab/chat-tutor-embeddable/pull/10